HX711.cpp

```diff
@@ -44,50 +44,48 @@ void HX711::set_gain(byte gain) {
 }
 
 long HX711::read() {
-	// wait for the chip to become ready
-	while (!is_ready()) {
-		// Will do nothing on Arduino but prevent resets of ESP8266 (Watchdog Issue)
-		yield();
-	}
-
-    unsigned long value = 0;
-    byte data[3] = { 0 };
-    byte filler = 0x00;
+    // Byte:     0        1        2        3
+    // Bits:  76543210 76543210 76543210 76543210
+    // Data: |--------|--------|--------|--------|
+    // Bit#:  33222222 22221111 11111100 00000000
+    //        10987654 32109876 54321098 76543210
+    union DataBuffer {
+        byte data[4];
+        long value;
+    } data_buffer;
+
+    // Wait for the chip to become ready
+    for (; !is_ready() ;) {
```

Comment:
It is nonsense. Arduino uses certain language and compiler is clever enough to optimise even more than you think. Don't waste readability for such purposes until it really needed.